### PR TITLE
Inbound transfer integration tests

### PR DESCRIPTION
**Pull request by truthfool**
- inbound transfer integration tests
Specifications : 
1. Savings account Creation Test
2. Loan account Creation Test
3. Deposit Savings Worker Test
@avikganguly01 @fynmanoj

### CODE REVIEW

File: src/test/java/org/mifos/integrationtest/cucumber/GSMATransferDef.java [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+
+    protected static RequestSpecification setHeadersLoan(RequestSpecification requestSpec) {
+        requestSpec.header("Fineract-Platform-TenantId", tenant);
+        requestSpec.header("Accept", "application/json, text/plain, */*");
+        requestSpec.header("Accept-Language", "en-GB,en-US;q=0.9,en;q=0.8");
+        requestSpec.header("Accept-Encoding", "gzip, deflate, br");
+        requestSpec.header("Authorization", "Basic bWlmb3M6cGFzc3dvcmQ=");
+        requestSpec.header("Origin", "https://65.0.42.17:8443");
+        requestSpec.header("Referer", "https://65.0.42.17:8443/");
+        requestSpec.header("Sec-Fetch-Dest", "empty");
```
Comment (fynmanoj) [inline src/test/java/org/mifos/integrationtest/cucumber/GSMATransferDef.java:101]: all these headers are not mandatory

Comment (fynmanoj) [reply src/test/java/org/mifos/integrationtest/cucumber/GSMATransferDef.java:101]: keep the minimal headers, teannt, auth and content type only.  also read tenant and auth from config

File: src/test/java/org/mifos/integrationtest/cucumber/GSMATransferDef.java [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+    }
+
+    protected static Object setBodyLoanRepayment(String amount) {
+        LoanRepaymentDTO loanRepaymentDTO = new LoanRepaymentDTO(currentDate, "1", amount, "en", "dd MMMM yyyy");
+        return loanRepaymentDTO.toString();
+    }
+
+    protected static RequestSpecification setHeadersSavings(RequestSpecification requestSpec) {
+        requestSpec.header("fineract-platform-tenantid", tenant);
+        requestSpec.header("accept", "application/json, text/plain, */*");
```
Comment (fynmanoj) [inline src/test/java/org/mifos/integrationtest/cucumber/GSMATransferDef.java:157]: use minimal and configurable headers. also reuse the methof

File: src/test/java/org/mifos/integrationtest/cucumber/GSMATransferDef.java [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
+        return savingsActivate.toString();
+    }
+
+    protected static String setSavingsDepositAccount(int amount) {
+        SavingsAccountDeposit savingsAccountDeposit = new SavingsAccountDeposit("dd MMMM yyyy", "en", 1, amount, currentDate);
+        return savingsAccountDeposit.toString();
+    }
+
+    protected static RequestSpecification setHeadersClient(RequestSpecification requestSpec) {
+        requestSpec.header("fineract-platform-tenantid", tenant);
```
Comment (fynmanoj) [inline src/test/java/org/mifos/integrationtest/cucumber/GSMATransferDef.java:223]: use minimal and configurable headers. also reuse the method

**Review by fynmanoj — CHANGES_REQUESTED** [review]

File: src/test/java/org/mifos/integrationtest/cucumber/GSMATransferDef.java (hunk 1 above)
Comment (truthfool) [reply src/test/java/org/mifos/integrationtest/cucumber/GSMATransferDef.java:101]: Tenant is read from feature file.

File: src/test/java/org/mifos/integrationtest/cucumber/GSMATransferDef.java (hunk 2 above)
Comment (truthfool) [reply src/test/java/org/mifos/integrationtest/cucumber/GSMATransferDef.java:157]: they have different letter case for headers that's why can'y reuse for loan and savings.

File: src/test/java/org/mifos/integrationtest/cucumber/GSMATransferDef.java (hunk 3 above)
Comment (truthfool) [reply src/test/java/org/mifos/integrationtest/cucumber/GSMATransferDef.java:223]: they have different letter case for headers that's why can'y reuse for loan and savings.

File: src/test/java/org/mifos/integrationtest/cucumber/GSMATransferDef.java (hunk 1 above)
Comment (fynmanoj) [reply src/test/java/org/mifos/integrationtest/cucumber/GSMATransferDef.java:98]: The origin and refferar IP address are not necessary

File: src/test/java/org/mifos/integrationtest/cucumber/GSMATransferDef.java [hunk 4]
```diff
@@ -0,0 +1,274 @@
...
+        tenant = tenantName;
+    }
+
+    protected void setcurrentDate(String date) {
+        currentDate = date;
+    }
+
+    protected RequestSpecification setHeadersLoan(RequestSpecification requestSpec) {
+        requestSpec.header("Fineract-Platform-TenantId", tenant);
+        requestSpec.header("Accept", "application/json, text/plain, */*");
```
Comment (fynmanoj) [inline src/test/java/org/mifos/integrationtest/cucumber/GSMATransferDef.java:92]: Remove this

File: src/test/java/org/mifos/integrationtest/cucumber/GSMATransferDef.java [hunk 5]
```diff
@@ -0,0 +1,274 @@
...
+        requestSpec.header("Accept-Language", "en-GB,en-US;q=0.9,en;q=0.8");
```
Comment (fynmanoj) [inline src/test/java/org/mifos/integrationtest/cucumber/GSMATransferDef.java:93]: Remove this

File: src/test/java/org/mifos/integrationtest/cucumber/GSMATransferDef.java [hunk 6]
```diff
@@ -0,0 +1,274 @@
...
+        requestSpec.header("Accept-Encoding", "gzip, deflate, br");
```
Comment (fynmanoj) [inline src/test/java/org/mifos/integrationtest/cucumber/GSMATransferDef.java:94]: Remove this

File: src/test/java/org/mifos/integrationtest/cucumber/GSMATransferDef.java [hunk 7]
```diff
@@ -0,0 +1,274 @@
...
+        requestSpec.header("Authorization", "Basic bWlmb3M6cGFzc3dvcmQ=");
```
Comment (fynmanoj) [inline src/test/java/org/mifos/integrationtest/cucumber/GSMATransferDef.java:95]: Remove everything except auth, content type and tenants

File: src/test/java/org/mifos/integrationtest/cucumber/GSMATransferDef.java [hunk 8]
```diff
@@ -0,0 +1,274 @@
...
+        LoanApprove loanApprove = new LoanApprove(currentDate, amount, currentDate, new ArrayList<>(), "en", "dd MMMM yyyy");
+        return objectMapper.writeValueAsString(loanApprove);
+    }
+
+    protected Object setBodyLoanRepayment(String amount) throws JsonProcessingException {
+        LoanRepaymentDTO loanRepaymentDTO = new LoanRepaymentDTO(currentDate, "1", amount, "en", "dd MMMM yyyy");
+        return objectMapper.writeValueAsString(loanRepaymentDTO);
+    }
+
+    protected RequestSpecification setHeadersSavings(RequestSpecification requestSpec) {
```
Comment (fynmanoj) [inline src/test/java/org/mifos/integrationtest/cucumber/GSMATransferDef.java:152]: Delete this method and re use the one given above

File: src/test/java/org/mifos/integrationtest/cucumber/GSMATransferDef.java (hunk 3 above)
Comment (fynmanoj) [reply src/test/java/org/mifos/integrationtest/cucumber/GSMATransferDef.java:222]: Http is not case sensitive. Try reusing

File: src/test/java/org/mifos/integrationtest/cucumber/GSMATransferDef.java (hunk 2 above)
Comment (fynmanoj) [reply src/test/java/org/mifos/integrationtest/cucumber/GSMATransferDef.java:154]: Http is not case sensitive, so try reusing

File: src/main/resources/application.yaml [hunk 9]
```diff
@@ -29,3 +30,27 @@ retry-interval: 15000
 
 defaults:
   tenant: "gorilla"
+
```
Comment (fynmanoj) [inline src/main/resources/application.yaml:33]: Add authentication also here as a parameter

Comment (fynmanoj) [inline src/main/resources/application.yaml:33]: Add authentication also here as a parameter

File: src/test/java/org/mifos/integrationtest/cucumber/GSMATransferDef.java (hunk 7 above)
Comment (fynmanoj) [inline src/test/java/org/mifos/integrationtest/cucumber/GSMATransferDef.java:95]: Take auth from a config yml

**Review by fynmanoj — CHANGES_REQUESTED** [review]
Remove additional headers and re-use code. Remove hard-coded authentication and move that to a configfile

File: src/test/java/org/mifos/integrationtest/cucumber/GSMATransferDef.java (hunk 8 above)
Comment (truthfool) [reply src/test/java/org/mifos/integrationtest/cucumber/GSMATransferDef.java:152]: both have different cases in headers

**Review by fynmanoj — APPROVED** [review]

**Pull request merged**